mk/spksrc.common.mk

```diff
@@ -13,13 +13,22 @@ MSG = echo "===> "
 # Launch command in the working dir of the package source and the right environment
 RUN = cd $(WORK_DIR)/$(PKG_DIR) && env $(ENV)
 
-# Pip command
+# fallback by default to native/python*
 PIP ?= pip
+
+# Cross compiling must call "pip" directly to ensure using
+# $(WORK_DIR)/crossenv pip version instead from adjusted $PATH
+PIP_CROSS = $(shell . $(CROSSENV) && $(RUN) which pip)
+
+# System default pip outside from build environment
+PIP_SYSTEM = $(shell which pip)
+
 # Why ask for the same thing twice? Always cache downloads
-PIP_CACHE_OPT ?= --cache-dir $(PIP_DIR)
+PIP_CACHE_OPT ?= --cache-dir $(PIP_DIR) --find-links $(BASE_DISTRIB_DIR)
+
+# Define default wheel & download pip options
 PIP_WHEEL_ARGS = wheel --no-binary :all: $(PIP_CACHE_OPT) --no-deps --wheel-dir $(WHEELHOUSE)
-PIP_WHEEL = $(PIP) $(PIP_WHEEL_ARGS)
-PIP_DOWNLOAD = $(PIP) download $(PIP_ARGS) --dest $(BASE_DISTRIB_DIR)
+PIP_DOWNLOAD_ARGS = download --no-binary :all: $(PIP_CACHE_OPT) --dest $(BASE_DISTRIB_DIR) --no-build-isolation
```

Comment:
While looking more closely to the logs I noticed another issue with this: the `--cache-dir` option.
When building in my own environment the caching works really well.
Although something I noticed in github-action environment, during the building process it shows this message:
```
WARNING: The directory '/github/workspace/distrib/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
```
So while having all the wheels.tar.bz2 files under `distrib` seems obvious to me (although if I understand you'd prefer it elsewhere?), the caching directory `distrib/pip` poses a problem currently.  @publicarray any clue why?

Reply:
That's unfortunate hm 🤔 How is the folder created? Maybe creating it manually before pip runs could give us a clue?

Yea something like `distrib/wheel_downloads`, `distrib/pip/cache` or something. I think keeping every cache / download separate is better for organising and cleaning out later (easy to remove all pip cache stuff if needed on a local machine for debugging). IMHO The root of distrib should only contain the downloaded archives for the cross/packages.

Reply:
As long as this PR didn't introduce this bug we can still merge this. I prefer Smaller specific PRs anyway.

Reply:
Revisiting this I presume the `distrib` folder gets mounted r+w at download stage then read-only at build stage (or with different user privileges). While being exceptional this would make sense as you might end-up getting concurrent writes into the pip cache directory once parallel-builds kicks-off.  I ain't had much exposure to this side of the fence so I may be wrong but that's my first assumption. 

Also lookint at other build logs this error ain't new (e.g. was not introduced with this PR)
```
9_Build (aarch64-7.0).txt:2021-11-18T22:51:28.0187288Z WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
9_Build (aarch64-7.0).txt:2021-11-18T22:51:36.4440129Z WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
9_Build (aarch64-7.0).txt:2021-11-18T22:58:02.8772144Z WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
9_Build (aarch64-7.0).txt:2021-11-18T22:58:03.5349151Z WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
```

It now makes me wonder if caching is needed at all?
```
# Why ask for the same thing twice? Always cache downloads
PIP_CACHE_OPT ?= --cache-dir $(PIP_DIR) --find-links $(BASE_DISTRIB_DIR)
```
Could be downsized to just:
```
# Why ask for the same thing twice? Always cache downloads
PIP_CACHE_OPT ?= --find-links $(BASE_DISTRIB_DIR)
```
Which is now then honored with the `--dest $(BASE_DISTRIB_DIR)` during download followed by being enforced using `--no-index` flag when building wheels to use the local file.

Reply:
pip download support is really bad.  I ended-up having to write a workaround that https://github.com/pypa/pip/issues/1884#issuecomment-997227879

Also caching is of no use with github-action so it's now disabled.